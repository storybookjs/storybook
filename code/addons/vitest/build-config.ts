import type { BuildEntries } from '../../../scripts/build/utils/entry-utils';

const config: BuildEntries = {
  entries: {
    browser: [
      {
        exportEntries: ['.'],
        entryPoint: './src/index.ts',
      },
      {
        exportEntries: ['./manager'],
        entryPoint: './src/manager.tsx',
        dts: false,
      },
      {
        exportEntries: ['./internal/setup-file'],
        entryPoint: './src/vitest-plugin/setup-file.ts',
        dts: false,
      },
      {
        exportEntries: ['./internal/test-utils'],
        entryPoint: './src/vitest-plugin/test-utils.ts',
        dts: false,
      },
    ],
    node: [
      {
        exportEntries: ['./preset'],
        entryPoint: './src/preset.ts',
        dts: false,
      },
      {
        exportEntries: ['./internal/global-setup'],
        entryPoint: './src/vitest-plugin/global-setup.ts',
        dts: false,
      },
      {
        exportEntries: ['./vitest'],
        entryPoint: './src/node/vitest.ts',
        dts: false,
      },
      {
        exportEntries: ['./postinstall'],
        entryPoint: './src/postinstall.ts',
        dts: false,
      },
      {
        exportEntries: ['./vitest-plugin'],
        entryPoint: './src/vitest-plugin/index.ts',
      },
      {
        exportEntries: ['./internal/coverage-reporter'],
        entryPoint: './src/node/coverage-reporter.ts',
        dts: false,
      },
    ],
  },
};

export default config;
