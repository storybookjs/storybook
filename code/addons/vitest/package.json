{
  "name": "@storybook/addon-vitest",
  "version": "9.1.0-alpha.7",
  "description": "Storybook addon for testing components",
  "keywords": [
    "storybook-addons",
    "addon-vitest",
    "vitest",
    "testing",
    "test"
  ],
  "homepage": "https://github.com/storybookjs/storybook/tree/next/code/addons/vitest",
  "bugs": {
    "url": "https://github.com/storybookjs/storybook/issues"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/storybookjs/storybook.git",
    "directory": "code/addons/vitest"
  },
  "funding": {
    "type": "opencollective",
    "url": "https://opencollective.com/storybook"
  },
  "license": "MIT",
  "type": "module",
  "imports": {
    "#manager-store": {
      "storybook": "./src/manager-store.mock.ts",
      "default": "./src/manager-store.ts"
    }
  },
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "default": "./dist/index.js"
    },
    "./internal/coverage-reporter": "./dist/node/coverage-reporter.js",
    "./internal/global-setup": "./dist/vitest-plugin/global-setup.js",
    "./internal/setup-file": "./dist/vitest-plugin/setup-file.js",
    "./internal/test-utils": "./dist/vitest-plugin/test-utils.js",
    "./manager": "./dist/manager.js",
    "./package.json": "./package.json",
    "./postinstall": "./dist/postinstall.js",
    "./preset": "./dist/preset.js",
    "./static/coverage-reporter.cjs": "./static/coverage-reporter.cjs",
    "./vitest": "./dist/node/vitest.js",
    "./vitest-plugin": {
      "types": "./dist/vitest-plugin/index.d.ts",
      "default": "./dist/vitest-plugin/index.js"
    }
  },
  "files": [
    "dist/**/*",
    "templates/**/*",
    "!dist/dummy.*",
    "README.md",
    "*.mjs",
    "*.js",
    "*.cjs",
    "*.d.ts",
    "!src/**/*"
  ],
  "scripts": {
    "check": "jiti ../../../scripts/check/check-package.ts",
    "prep": "jiti ../../../scripts/build/build-package.ts"
  },
  "dependencies": {
    "@storybook/global": "^5.0.0",
    "@storybook/icons": "^1.4.0",
    "prompts": "^2.4.0",
    "ts-dedent": "^2.2.0"
  },
  "devDependencies": {
    "@types/istanbul-lib-report": "^3.0.3",
    "@types/micromatch": "^4.0.0",
    "@types/node": "^22.0.0",
    "@types/semver": "^7",
    "@vitest/browser": "^3.2.4",
    "@vitest/runner": "^3.2.4",
    "boxen": "^8.0.1",
    "es-toolkit": "^1.36.0",
    "execa": "^8.0.1",
    "find-up": "^7.0.0",
    "istanbul-lib-report": "^3.0.1",
    "micromatch": "^4.0.8",
    "pathe": "^1.1.2",
    "picocolors": "^1.1.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "semver": "^7.6.3",
    "sirv": "^2.0.4",
    "slash": "^5.0.0",
    "tinyglobby": "^0.2.10",
    "tree-kill": "^1.2.2",
    "ts-dedent": "^2.2.0",
    "typescript": "^5.8.3",
    "vitest": "^3.2.4"
  },
  "peerDependencies": {
    "@vitest/browser": "^3.0.0",
    "@vitest/runner": "^3.0.0",
    "storybook": "workspace:^",
    "vitest": "^3.0.0"
  },
  "peerDependenciesMeta": {
    "@vitest/browser": {
      "optional": true
    },
    "@vitest/runner": {
      "optional": true
    },
    "vitest": {
      "optional": true
    }
  },
  "publishConfig": {
    "access": "public"
  },
  "storybook": {
    "displayName": "Test",
    "unsupportedFrameworks": [
      "react-native"
    ],
    "icon": "https://user-images.githubusercontent.com/263385/101991666-479cc600-3c7c-11eb-837b-be4e5ffa1bb8.png"
  }
}
