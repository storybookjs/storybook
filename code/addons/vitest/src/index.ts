import { definePreview } from 'storybook/preview-api';

export default () => definePreview({});

export type { TestParameters } from './types';
