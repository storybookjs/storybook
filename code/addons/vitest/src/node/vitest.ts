import { createRequire } from 'node:module';
import process from 'node:process';

import { Channel } from 'storybook/internal/channels';

import {
  ADDON_ID,
  STATUS_TYPE_ID_A11Y,
  STATUS_TYPE_ID_COMPONENT_TEST,
  storeOptions,
} from '../constants';
import type { ErrorLike, FatalErrorEvent, StoreEvent, StoreState } from '../types';
import { TestManager } from './test-manager';

const require = createRequire(import.meta.url);

// we need to require core-server here, because its ESM output is not valid
const {
  experimental_UniversalStore: UniversalStore,
  experimental_getStatusStore: getStatusStore,
  experimental_getTestProviderStore: getTestProviderStore,
} = require('storybook/internal/core-server') as {
  experimental_UniversalStore: typeof import('storybook/internal/core-server').experimental_UniversalStore;
  experimental_getStatusStore: typeof import('storybook/internal/core-server').experimental_getStatusStore;
  experimental_getTestProviderStore: typeof import('storybook/internal/core-server').experimental_getTestProviderStore;
};

const channel: Channel = new Channel({
  async: true,
  transport: {
    send: (event) => {
      process.send?.(event);
    },
    setHandler: (handler) => {
      process.on('message', handler);
    },
  },
});

// eslint-disable-next-line no-underscore-dangle
(UniversalStore as any).__prepare(channel, UniversalStore.Environment.SERVER);

const store = UniversalStore.create<StoreState, StoreEvent>(storeOptions);

new TestManager({
  store,
  componentTestStatusStore: getStatusStore(STATUS_TYPE_ID_COMPONENT_TEST),
  a11yStatusStore: getStatusStore(STATUS_TYPE_ID_A11Y),
  testProviderStore: getTestProviderStore(ADDON_ID),
  onReady: () => {
    process.send?.({ type: 'ready' });
  },
});

const exit = (code = 0) => {
  channel?.removeAllListeners();
  process.exit(code);
};

const createUnhandledErrorHandler = (message: string) => async (error: ErrorLike) => {
  try {
    const payload: FatalErrorEvent['payload'] = {
      message,
      error: {
        message: error.message,
        name: error.name,
        stack: error.stack,
        cause: error.cause as ErrorLike,
      },
    };
    // Node.js will exit immediately in these situations, so we can't send an event via the universal store
    // because the process will exit before the event is sent.
    // we're sending it manually instead, so the parent process can forward it to the store.
    process.send?.({
      type: 'uncaught-error',
      payload,
    });
  } finally {
    exit(1);
  }
};

process.on(
  'uncaughtException',
  createUnhandledErrorHandler('Uncaught exception in the test runner process')
);
process.on(
  'unhandledRejection',
  createUnhandledErrorHandler('Unhandled rejection in the test runner process')
);

process.on('exit', exit);
process.on('SIGINT', () => exit(0));
process.on('SIGTERM', () => exit(0));
