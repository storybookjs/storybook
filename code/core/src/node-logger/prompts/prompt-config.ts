import type { PromptProvider } from './prompt-provider-base';
import { ClackPromptProvider } from './prompt-provider-clack';
import { PromptsPromptProvider } from './prompt-provider-prompts';

type PromptLibrary = 'clack' | 'prompts';

const PROVIDERS = {
  clack: new ClackPromptProvider(),
  prompts: new PromptsPromptProvider(),
} as const;

let currentPromptLibrary: PromptLibrary = process.env.USE_CLACK === 'true' ? 'clack' : 'prompts';

export const setPromptLibrary = (library: PromptLibrary): void => {
  currentPromptLibrary = library;
};

export const getPromptLibrary = (): PromptLibrary => {
  return currentPromptLibrary;
};

export const getPromptProvider = (): PromptProvider => {
  return PROVIDERS[currentPromptLibrary];
};

export const isClackEnabled = (): boolean => {
  return currentPromptLibrary === 'clack';
};

export const isPromptsEnabled = (): boolean => {
  return currentPromptLibrary === 'prompts';
};

/**
 * Returns the preferred stdio for the current prompt library.
 *
 * Clack handles stdio output so that it clears it out later, and therefore 'pipe' is better. For
 * prompts, we want to inherit the stdio so that the output is displayed in the terminal.
 */
export const getPreferredStdio = (): 'inherit' | 'pipe' => {
  return isClackEnabled() ? 'pipe' : 'inherit';
};
