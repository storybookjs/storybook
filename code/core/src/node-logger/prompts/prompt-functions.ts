import { wrapTextForClack, wrapTextForClackHint } from '../wrap-utils';
import { getPromptProvider } from './prompt-config';
import type {
  BasePromptOptions,
  ConfirmPromptOptions,
  MultiSelectPromptOptions,
  Option,
  PromptOptions,
  SelectPromptOptions,
  SpinnerInstance,
  SpinnerOptions,
  TaskLogInstance,
  TaskLogOptions,
  TextPromptOptions,
} from './prompt-provider-base';
import { asyncLocalStorage } from './storage';

// Re-export types for convenience
export type {
  Option,
  BasePromptOptions,
  TextPromptOptions,
  ConfirmPromptOptions,
  SelectPromptOptions,
  MultiSelectPromptOptions,
  PromptOptions,
  SpinnerInstance,
  TaskLogInstance,
  TaskLogOptions,
};

// Global state for tracking active spinners and task logs
let activeSpinner: SpinnerInstance | null = null;
let activeTaskLog: TaskLogInstance | null = null;
let originalConsoleLog: typeof console.log | null = null;

// Console.log patching functions
const patchConsoleLog = () => {
  if (!originalConsoleLog) {
    originalConsoleLog = console.log;
    console.log = (...args: any[]) => {
      const message = args
        .map((arg) => (typeof arg === 'string' ? arg : JSON.stringify(arg)))
        .join(' ');

      if (activeTaskLog) {
        activeTaskLog.message(message);
      } else if (activeSpinner) {
        activeSpinner.message(message);
      } else {
        originalConsoleLog!(...args);
      }
    };
  }
};

const restoreConsoleLog = () => {
  if (originalConsoleLog && !activeSpinner && !activeTaskLog) {
    console.log = originalConsoleLog;
    originalConsoleLog = null;
  }
};

export const text = async (
  options: TextPromptOptions,
  promptOptions?: PromptOptions
): Promise<string> => {
  return getPromptProvider().text(options, promptOptions);
};

export const confirm = async (
  options: ConfirmPromptOptions,
  promptOptions?: PromptOptions
): Promise<boolean> => {
  return getPromptProvider().confirm(options, promptOptions);
};

export const select = async <T>(
  options: SelectPromptOptions<T>,
  promptOptions?: PromptOptions
): Promise<T> => {
  return getPromptProvider().select(options, promptOptions);
};

export const multiselect = async <T>(
  options: MultiSelectPromptOptions<T>,
  promptOptions?: PromptOptions
): Promise<T[]> => {
  return getPromptProvider().multiselect(
    {
      ...options,
      options: options.options.map((opt) => ({
        ...opt,
        hint: opt.hint
          ? wrapTextForClackHint(opt.hint, undefined, opt.label || String(opt.value))
          : undefined,
      })),
    },
    promptOptions
  );
};

export const spinner = (options: SpinnerOptions): SpinnerInstance => {
  const spinnerInstance = getPromptProvider().spinner(options);

  // Wrap the spinner methods to handle console.log patching
  const wrappedSpinner: SpinnerInstance = {
    start: (message?: string) => {
      activeSpinner = wrappedSpinner;
      patchConsoleLog();
      spinnerInstance.start(message);
    },
    stop: (message?: string) => {
      activeSpinner = null;
      restoreConsoleLog();
      spinnerInstance.stop(message);
    },
    message: (text: string) => {
      spinnerInstance.message(text);
    },
  };

  return wrappedSpinner;
};

export const taskLog = (options: TaskLogOptions): TaskLogInstance => {
  const task = getPromptProvider().taskLog(options);

  // Wrap the task log methods to handle console.log patching
  const wrappedTaskLog: TaskLogInstance = {
    message: (message: string) => {
      task.message(wrapTextForClack(message));
    },
    success: (message: string, options?: { showLog?: boolean }) => {
      activeTaskLog = null;
      restoreConsoleLog();
      task.success(message, options);
    },
    error: (message: string) => {
      activeTaskLog = null;
      restoreConsoleLog();
      task.error(message);
    },
  };

  // Activate console.log patching when task log is created
  activeTaskLog = wrappedTaskLog;
  patchConsoleLog();

  return wrappedTaskLog;
};
