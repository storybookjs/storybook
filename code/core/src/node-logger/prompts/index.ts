import { executeTask, executeTaskWithSpinner } from '../tasks';
import * as promptConfig from './prompt-config';
import * as promptFunctions from './prompt-functions';

export const prompt = {
  ...promptFunctions,
  ...promptConfig,
  executeTask,
  executeTaskWithSpinner,
};
