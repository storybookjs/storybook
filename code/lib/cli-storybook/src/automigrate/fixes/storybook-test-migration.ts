import { readFile, writeFile } from 'node:fs/promises';

import picocolors from 'picocolors';
import prompts from 'prompts';
import { dedent } from 'ts-dedent';

import type { Fix } from '../types';

interface StorybookTestMigrationResult {
  hasDependency: boolean;
  defaultGlob: string;
}

export const storybookTestMigration: Fix<StorybookTestMigrationResult> = {
  id: 'storybook-test-migration',
  versionRange: ['<9.0.0', '>=9.0.0-0'],
  check: async ({ packageManager, configDir }) => {
    const storybookTestPackageVersion = await packageManager.getPackageVersion('@storybook/test');

    if (!storybookTestPackageVersion) {
      return null;
    }

    // Default glob pattern to scan for files
    const defaultGlob = `{${configDir}/**/*,**/*.{stories.*,test.*}}`;

    return {
      hasDependency: !!storybookTestPackageVersion,
      defaultGlob,
    };
  },
  prompt: (result) => {
    return dedent`
      We detected that you're using ${picocolors.magenta('@storybook/test')} in your project. This package has been merged into ${picocolors.magenta('storybook/test')}.
      
      We'll help you migrate by:
      1. Removing ${picocolors.magenta('@storybook/test')} from your dependencies
      2. Updating all imports from ${picocolors.magenta('@storybook/test')} to ${picocolors.magenta('storybook/test')}
      
      We'll scan the following files:
      - All files in .storybook directory
      - All *.stories.* files
      - All *.test.* files
      
      The default glob pattern we'll use is: ${picocolors.yellow(result.defaultGlob)}
      
      In the next step, you can provide a custom glob pattern to fine-tune the files we'll scan.
    `;
  },
  promptType: 'auto',
  run: async ({ packageManager, dryRun, result }) => {
    // Remove @storybook/test from dependencies
    if (!dryRun) {
      await packageManager.removeDependencies({}, ['@storybook/test']);
    }

    // Find all files matching the glob pattern
    const { glob } = await prompts({
      type: 'text',
      name: 'glob',
      message: 'Enter a custom glob pattern to scan (or press enter to use default):',
      initial: result.defaultGlob,
    });

    // eslint-disable-next-line depend/ban-dependencies
    const globby = (await import('globby')).globby;

    const files = await globby(glob, {
      ignore: ['node_modules/**', 'dist/**', 'build/**'],
      cwd: process.cwd(),
      gitignore: true,
    });

    let updatedFilesCount = 0;

    // Process each file
    for (const file of files) {
      const content = await readFile(file, 'utf-8');

      // Replace imports
      const updatedContent = content.replace(/(['"])@storybook\/test\1/g, '$1storybook/test$1');

      if (content !== updatedContent) {
        if (!dryRun) {
          await writeFile(file, updatedContent);
        }
        // Track updated files count to show at the end
        updatedFilesCount = (updatedFilesCount || 0) + 1;
      }
    }

    console.log(`Updated ${updatedFilesCount} files`);
  },
};
