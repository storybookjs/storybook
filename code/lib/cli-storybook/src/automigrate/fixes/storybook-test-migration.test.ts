import { readFile, writeFile } from 'node:fs/promises';

import { beforeEach, describe, expect, it, vi } from 'vitest';

import type { JsPackageManager } from 'storybook/internal/common';

import prompts from 'prompts';

import { storybookTestMigration } from './storybook-test-migration';

// Mock fs promises
vi.mock('node:fs/promises', () => ({
  readFile: vi.fn(),
  writeFile: vi.fn(),
}));

// Mock prompts
vi.mock('prompts', () => ({
  default: vi.fn(),
}));

// Mock globby
vi.mock('globby', () => ({
  globby: vi.fn(),
}));

vi.mock('picocolors', () => ({
  default: {
    magenta: vi.fn().mockImplementation((text) => text),
    yellow: vi.fn().mockImplementation((text) => text),
  },
}));

describe('storybook-test-migration', () => {
  const mockPackageManager = {
    getPackageVersion: vi.fn(),
    removeDependencies: vi.fn(),
    type: 'npm',
    getPackageJson: vi.fn(),
  } as unknown as JsPackageManager;

  const mockConfigDir = '.storybook';

  beforeEach(() => {
    vi.clearAllMocks();
  });

  it('should return null if @storybook/test is not in dependencies', async () => {
    vi.mocked(mockPackageManager.getPackageVersion).mockResolvedValue(null);

    const result = await storybookTestMigration.check({
      packageManager: mockPackageManager,
      configDir: mockConfigDir,
    } as any);

    expect(result).toBeNull();
  });

  it('should return migration result if @storybook/test is in dependencies', async () => {
    vi.mocked(mockPackageManager.getPackageVersion).mockResolvedValue('^7.0.0');

    const result = await storybookTestMigration.check({
      packageManager: mockPackageManager,
      configDir: mockConfigDir,
    } as any);

    expect(result).toEqual({
      hasDependency: true,
      defaultGlob: '{.storybook/**/*,**/*.{stories.*,test.*}}',
    });
  });

  it('should show the correct prompt message', () => {
    const result = {
      hasDependency: true,
      defaultGlob: '{.storybook/**/*,**/*.{stories.*,test.*}}',
    };

    const prompt = storybookTestMigration.prompt(result);
    expect(prompt).toMatchSnapshot();
  });

  it('should remove @storybook/test dependency and update imports', async () => {
    // Mock package.json
    vi.mocked(mockPackageManager.getPackageVersion).mockResolvedValue('^7.0.0');

    // Mock globby to return test files
    // eslint-disable-next-line depend/ban-dependencies
    const { globby } = await import('globby');
    (globby as any).mockResolvedValue(['test1.ts', 'test2.ts']);

    // Mock prompts to return default glob
    (prompts as any).mockResolvedValue({ glob: '{.storybook/**/*,**/*.{stories.*,test.*}}' });

    // Mock file contents
    (readFile as any).mockResolvedValue('import { expect } from "@storybook/test";');

    // Run the migration
    await storybookTestMigration.run?.({
      packageManager: mockPackageManager,
      dryRun: false,
      result: {
        hasDependency: true,
        defaultGlob: '{.storybook/**/*,**/*.{stories.*,test.*}}',
      },
    } as any);

    // Verify dependency removal
    expect(mockPackageManager.removeDependencies).toHaveBeenCalledWith({}, ['@storybook/test']);

    // Verify file updates
    expect(writeFile).toHaveBeenCalledWith('test1.ts', 'import { expect } from "storybook/test";');
    expect(writeFile).toHaveBeenCalledWith('test2.ts', 'import { expect } from "storybook/test";');
  });

  it('should handle dry run mode', async () => {
    // Mock package.json
    vi.mocked(mockPackageManager.getPackageVersion).mockResolvedValue('^7.0.0');

    // Mock globby to return test files
    // eslint-disable-next-line depend/ban-dependencies
    const { globby } = await import('globby');
    (globby as any).mockResolvedValue(['test1.ts']);

    // Mock prompts to return default glob
    (prompts as any).mockResolvedValue({ glob: '{.storybook/**/*,**/*.{stories.*,test.*}}' });

    // Mock file contents
    (readFile as any).mockResolvedValue('import { expect } from "@storybook/test";');

    // Run the migration in dry run mode
    await storybookTestMigration.run?.({
      packageManager: mockPackageManager,
      dryRun: true,
      result: {
        hasDependency: true,
        defaultGlob: '{.storybook/**/*,**/*.{stories.*,test.*}}',
      },
    } as any);

    // Verify no changes were made
    expect(mockPackageManager.removeDependencies).not.toHaveBeenCalled();
    expect(writeFile).not.toHaveBeenCalled();
  });

  it('should handle different quote styles in imports', async () => {
    // Mock package.json
    vi.mocked(mockPackageManager.getPackageVersion).mockResolvedValue('^7.0.0');

    // Mock globby to return test files
    // eslint-disable-next-line depend/ban-dependencies
    const { globby } = await import('globby');
    (globby as any).mockResolvedValue(['test1.ts']);

    // Mock prompts to return default glob
    (prompts as any).mockResolvedValue({ glob: '{.storybook/**/*,**/*.{stories.*,test.*}}' });

    // Mock file contents with different quote styles
    (readFile as any).mockResolvedValue(
      'import { expect } from "@storybook/test";\nimport { expect } from \'@storybook/test\';'
    );

    // Run the migration
    await storybookTestMigration.run?.({
      packageManager: mockPackageManager,
      dryRun: false,
      result: {
        hasDependency: true,
        defaultGlob: '{.storybook/**/*,**/*.{stories.*,test.*}}',
      },
    } as any);

    // Verify quote styles were preserved
    expect(writeFile).toHaveBeenCalledWith(
      'test1.ts',
      'import { expect } from "storybook/test";\nimport { expect } from \'storybook/test\';'
    );
  });
});
