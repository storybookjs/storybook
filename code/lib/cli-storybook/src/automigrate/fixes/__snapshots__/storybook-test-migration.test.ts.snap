// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`storybook-test-migration > should show the correct prompt message 1`] = `
"We detected that you're using @storybook/test in your project. This package has been merged into storybook/test.

We'll help you migrate by:
1. Removing @storybook/test from your dependencies
2. Updating all imports from @storybook/test to storybook/test

We'll scan the following files:
- All files in .storybook directory
- All *.stories.* files
- All *.test.* files

The default glob pattern we'll use is: {.storybook/**/*,**/*.{stories.*,test.*}}

In the next step, you can provide a custom glob pattern to fine-tune the files we'll scan."
`;
